src/main/webapp/app/iris/manage/settings/shared/iris-enabled/iris-enabled.component.ts

```diff
@@ -75,4 +138,37 @@ export class IrisEnabledComponent implements OnInit {
                 break;
         }
     }
+
+    private handleSubSettingsTypeAll() {
+        const chatSettings = this.irisSettings?.irisChatSettings;
+        const textExerciseChatSettings = this.irisSettings?.irisTextExerciseChatSettings;
+        const courseChatSettings = this.irisSettings?.irisCourseChatSettings;
+        const competencyGenerationSettings = this.irisSettings?.irisCompetencyGenerationSettings;
+        const lectureIngestionSettings = this.irisSettings?.irisLectureIngestionSettings;
+        const faqIngestionSettings = this.irisSettings?.irisFaqIngestionSettings;
+        const lectureSettings = this.irisSettings?.irisLectureChatSettings;
+        this.irisSubSettings = chatSettings ? { ...chatSettings } : { type: IrisSubSettingsType.CHAT, enabled: false };
+
+        const allSubSettingsEnabled =
+            chatSettings?.enabled &&
+            textExerciseChatSettings?.enabled &&
+            courseChatSettings?.enabled &&
+            competencyGenerationSettings?.enabled &&
+            lectureIngestionSettings?.enabled &&
+            faqIngestionSettings?.enabled &&
+            lectureSettings?.enabled;
+
+        const anySubSettingEnabled = [
+            chatSettings?.enabled,
+            textExerciseChatSettings?.enabled,
+            courseChatSettings?.enabled,
+            competencyGenerationSettings?.enabled,
+            lectureIngestionSettings?.enabled,
+            faqIngestionSettings?.enabled,
+            lectureSettings?.enabled,
+        ].some((settingEnabled) => settingEnabled === true);
+
+        this.someButNotAllSettingsEnabled = anySubSettingEnabled && !allSubSettingsEnabled;
+        this.irisSubSettings.enabled = anySubSettingEnabled;
+    }
```

Comment:
I suggest the variant below which is a little more concise (roughly half as long)

```suggestion
    private handleSubSettingsTypeAll() {
        const subSettings = [
            this.irisSettings?.irisChatSettings,
            this.irisSettings?.irisTextExerciseChatSettings,
            this.irisSettings?.irisCourseChatSettings,
            this.irisSettings?.irisCompetencyGenerationSettings,
            this.irisSettings?.irisLectureIngestionSettings,
            this.irisSettings?.irisFaqIngestionSettings,
            this.irisSettings?.irisLectureChatSettings,
        ];

        const allEnabled = subSettings.every(s => s?.enabled);
        const anyEnabled = subSettings.some(s => s?.enabled);

        this.irisSubSettings = { type: IrisSubSettingsType.ALL, enabled: anyEnabled };
        this.someButNotAllSettingsEnabled = anyEnabled && !allEnabled;
    }
```